samples/dnn/text_detection.cpp

```diff
@@ -159,3 +215,16 @@ void decode(const Mat& scores, const Mat& geometry, float scoreThresh,
         }
     }
 }
+
+void decode_text(Mat& scores, std::string& text) {
+    String alphabet = "0123456789abcdefghijklmnopqrstuvwxyz";
+    for (int symbol_index = 0; symbol_index < scores.size[0]; ++symbol_index) {
+        Mat slice = Mat(1, scores.size[2], CV_32F, scores.ptr<float>(symbol_index)); // 37 one-hot encoded pins
+        Point p;
+        double m;
+        minMaxLoc(slice, 0, &m, 0, &p);
```

Comment:
Intead of multiple `minMaxLoc` it's better to use [cv::reduce](https://docs.opencv.org/master/d2/de8/group__core__array.html#ga4b78072a303f29d9031d56e5638da78e) with `REDUCE_MAX`.

Reply:
We cannot use cv::reduce here because we need a character index.

Reply:
Looks like it's very good feature to have. If you want, you may propose a feature with `REDUCE_MAX_IDX` for `cv::reduce` in the separate PR. (corresponding issue https://github.com/opencv/opencv/issues/10286)